client.go

```diff
@@ -210,16 +258,101 @@ func (client *Client) Do(req Req) (Res, error) {
 			} else if httpRes.StatusCode == 408 || (httpRes.StatusCode >= 501 && httpRes.StatusCode <= 599) {
 				log.Printf("[ERROR] HTTP Request failed: StatusCode %v, Retries: %v", httpRes.StatusCode, attempts)
 				continue
+			} else if httpRes.StatusCode == 401 && strings.Contains(res.Get("error").String(), "token has expired") {
+				log.Printf("[ERROR] HTTP Request failed: StatusCode %v, Retries: %v", httpRes.StatusCode, attempts)
+				client.Token = ""
+				err := client.Authenticate()
+				if err != nil {
+					log.Printf("[ERROR] Authentication failed: StatusCode %v, Retries: %v", httpRes.StatusCode, attempts)
+					log.Printf("[DEBUG] Exit from Do method")
+					return res, fmt.Errorf("HTTP Request failed: StatusCode %v", httpRes.StatusCode)
+				}
 			} else {
 				log.Printf("[ERROR] HTTP Request failed: StatusCode %v", httpRes.StatusCode)
+				log.Printf("[ERROR] HTTP Request failed: Full Response %v", httpRes)
 				log.Printf("[DEBUG] Exit from Do method")
-				return res, fmt.Errorf("HTTP Request failed: StatusCode %v", httpRes.StatusCode)
+				return res, fmt.Errorf("HTTP Request failed: StatusCode %v, Body: %v", httpRes.StatusCode, res)
```

Comment:
We introduced the `LogPayload` option to only print the full body when desired/safe. Wouldn't this unconditionally log the full payload?